central/sensor/service/pipeline/deploymentevents/pipeline.go

```diff
@@ -113,6 +114,9 @@ func (s *pipelineImpl) Run(ctx context.Context, clusterID string, msg *central.M
 	deployment := event.GetDeployment()
 	deployment.ClusterId = clusterID
 
+	// ROX-22002: Remove invalid null characters in annotations
+	deployment.Annotations = stringutils.SanitizeMapValues(deployment.GetAnnotations())
```

Comment:
Nit: I don't think we need to assign the map again right? maps are passed by reference so a deletion from a passed map is sufficient

Reply:
The function itself is not in place update.